edb/pgsql/deltadbops.py

```diff
@@ -206,10 +206,12 @@ def get_trigger_proc_text(self):
                           SCHEMA = '{schemaname}',
                           CONSTRAINT = '{constr}',
                           MESSAGE = '{errmsg}',
-                          DETAIL = 'Key ({plain_expr}) already exists.';
+                          DETAIL = 'Key ({esc_plain_expr}) already exists.';
                 END IF;
             '''.format(
                 plain_expr=origin_exprdata['plain'],
+                # XXX: do we want plain_expr here at all??
+                esc_plain_expr=origin_exprdata['plain'].replace("'", "''"),
```

Comment:
Use `pgsql.common.quote_literal` instead and concatenate `DETAIL` instead of interpolating directly.